agents/MarriageAgent.java

```diff
@@ -0,0 +1,78 @@
+package grakn.simulation.agents;
+
+import grakn.client.GraknClient.Transaction;
+import grakn.client.answer.ConceptMap;
+import grakn.simulation.common.RandomSource;
+import graql.lang.Graql;
+import graql.lang.query.GraqlGet;
+import graql.lang.query.GraqlInsert;
+import graql.lang.statement.Statement;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+public class MarriageAgent implements CityAgent {
+
+    private AgentContext context;
+
+    @Override
+    public void iterate(AgentContext context, RandomSource randomSource, World.City city) {
+
+        this.context = context;
+
+        // Find bachelors and bachelorettes who are considered adults and who are not in a marriage and pair them off randomly
+        Transaction tx = context.getGraknSession().transaction().write();
```

Comment:
Let's wrap the code using the `try` block:

```
try (Transaction tx = context.getGraknSession().transaction().write()) {
  ...
  if (numMarriagesPossible > 0) {
    ...
    tx.commit();
  }
}

It handles one edge case which is if an exception is thrown, the transaction is still properly closed.

Reply:
Coooool, added